tools/test-drive/src/sanity_checks.rs

```diff
@@ -0,0 +1,121 @@
+//! Module that contains sanity checks that Stable MIR APIs don't crash and that
+//! their result is coherent.
+//!
+//! These checks should only depend on StableMIR APIs. See other modules for tests that compare
+//! the result between StableMIR and internal APIs.
+use crate::TestResult;
+use rustc_smir::stable_mir;
+use std::fmt::Debug;
+use std::hint::black_box;
+
+fn check_equal<T>(val: T, expected: T, msg: &str) -> TestResult
+where
+    T: Debug + PartialEq,
+{
+    if val != expected {
+        Err(format!(
+            "{}: \n Expected: {:?}\n Found: {:?}",
+            msg, expected, val
+        ))
+    } else {
+        Ok(())
+    }
+}
+
+pub fn check(val: bool, msg: String) -> TestResult {
+    if !val {
+        Err(msg)
+    } else {
+        Ok(())
+    }
+}
+
+// Test that if there is an entry point, the function is part of `all_local_items`.
+pub fn test_entry_fn() -> TestResult {
+    let entry_fn = stable_mir::entry_fn();
+    entry_fn.map_or(Ok(()), |entry_fn| {
+        check_body(entry_fn.body());
+        let all_items = stable_mir::all_local_items();
+        check(
+            all_items.contains(&entry_fn),
+            format!("Failed to find entry_fn `{:?}`", entry_fn),
+        )
+    })
+}
+
+/// Check that the crate isn't empty and iterate over function bodies.
+pub fn test_all_fns() -> TestResult {
+    let all_items = stable_mir::all_local_items();
+    check(
+        !all_items.is_empty(),
+        "Failed to find any local item".to_string(),
+    )?;
+
+    // Not sure which API to use to make sure this is a function that has a body.
```

Comment:
outdated comment?

Reply:
True, I forgot to enable the test after you fixed the API

Reply:
Done